src/videopython/base/effects.py

```diff
@@ -98,3 +98,43 @@ def _apply(self, video: Video) -> Video:
             raise ValueError(f"Unknown mode: `{self.mode}`.")
         video.frames = np.asarray(new_frames)
         return video
+
+
+class Zoom(Effect):
+    def __init__(self, zoom_factor: float, mode: Literal["in", "out"]):
+        if zoom_factor <= 1:
+            raise ValueError("Zoom factor must be greater than 1!")
+        self.zoom_factor = zoom_factor
+        self.mode = mode
+
+    def _apply(self, video: Video) -> Video:
+        n_frames = len(video.frames)
+        new_frames = []
+
+        width = video.metadata.width
+        height = video.metadata.height
+        crop_sizes_w, crop_sizes_h = np.linspace(width // self.zoom_factor, width, n_frames), np.linspace(
+            height // self.zoom_factor, height, n_frames
+        )
+
+        if self.mode == "in":
+            for frame, w, h in tqdm(zip(video.frames, reversed(crop_sizes_w), reversed(crop_sizes_h))):
+
+                x = width / 2 - w / 2
+                y = height / 2 - h / 2
+
+                cropped_frame = frame[int(y) : int(y + h), int(x) : int(x + w)]
```

Comment:
Should we round them instead of int casting? Same in L134
```suggestion
                cropped_frame = frame[round(y) : round(y + h), round(x) : round(x + w)]
```

Reply:
Sure, we can do this.